<launch>

  <!-- Arguments -->
  <arg name="perception_goal" default="/waste_detector/goal"/>
  <arg name="odom" default="/local_odom"/>
  <arg name="status_topic" default="/agent_status"/>
  <arg name="dist_threshold" default="2.0"/>
  <arg name="goal_bounds_threshold" default="7.0"/>
  <arg name="config_file" default="grid_small.yaml" doc="YAML configuration file with waste positions"/>



  <!-- Waypoint Publisher -->
  <node pkg="kingfisher_experiments" type="waypoint_publisher.py" name="waypoint_publisher" output="screen">
    <remap from="~perception_goal" to="$(arg perception_goal)"/>
    <remap from="~odom" to="$(arg odom)"/>
    <remap from="~status_topic" to="$(arg status_topic)"/>
    <param name="dist_threshold" value="$(arg dist_threshold)"/>
    <param name="goal_bounds_threshold" value="$(arg goal_bounds_threshold)"/>
    <param name="config_file" value="$(arg config_file)"/>
  </node>

</launch>
